File: app/src/main/java/com/tonitealive/app/internal/di/modules/ActivityModule.java
package com.tonitealive.app.internal.di.modules;

import android.app.Activity;

import com.tonitealive.app.internal.di.annotations.PerActivity;

import dagger.Module;
import dagger.Provides;

@Module
public class ActivityModule {

    private final Activity activity;

    public ActivityModule(Activity activity) {
        this.activity = activity;
    }

    @Provides
    @PerActivity
    public Activity provideActivity() {
        return activity;
    }

}

File: app/src/main/kotlin/com/tonitealive/app/internal/di/modules/ActivityModule.kt
package com.tonitealive.app.internal.di.modules

import android.app.Activity
import com.tonitealive.app.internal.di.annotations.PerActivity
import dagger.Module
import dagger.Provides

@Module
open class ActivityModule(private val activity: Activity) {

    @Provides
    @PerActivity
    fun provideActivity(): Activity {
        return activity
    }

}